.github/workflows/build_and_test.yaml

```diff
@@ -52,8 +52,9 @@ jobs:
     - name: Run Coverage Tests
       run: make go.test.coverage
     - name: Upload coverage to Codecov
-      uses: codecov/codecov-action@4fe8c5f003fae66aa5ebb77cfd3e7bfbbda0b6b0  # v3.1.5
+      uses: codecov/codecov-action@e0b68c6749509c5f83f984dd99a76a1c1a231044  # v4.0.1
       with:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
how did this token get injected into secrets ?

Reply:
i have no idea why it passed now, maybe  there is a org secret.

Reply:
i would assume that no password is being passed - i think that is standard for opensource/public/prs - and dont see any secret defined anywhere

Reply:
we do have the codecov app installed for the repo, which maybe related

Reply:
yeah - its installed as an app - so it gets notifications and probs ci results from that